### ISSUE TITLE
keyattr dynamic or AttributeError

### ISSUE THREAD

**Issue by Loreton** [fabiocaccamo/python-benedict#261]
First of all I would like to renew my compliments and thanks for all the work you have done and continue to do in the creation of this tool which is becoming more and more interesting and useful in managing python dictionaries.

My current request focuses on the new dotted notation feature.

**_Example.1_**
```
>>> from benedict import benedict
>>> mydict=benedict(keyattr_enabled=True)
>>> mydict.ciao
{}
```

_in this case we have a dynamic creation of the "ciao" attribute_  

**_Example.2_**
```
>>> mydict._ciao
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/loreto/.local/lib/python3.10/site-packages/benedict/dicts/keyattr/keyattr_dict.py", line 27, in __getattr__
    raise AttributeError(
AttributeError: 'benedict' object has no attribute '_ciao'
```
_in this case we have a "_ciao" AttributeError... (why?)_

I don't know if the behavior shown in the above example is intended or not.

I would like to ask if it is possible to offer the possibility to choose what to do in case of AttributeError.
For example I might suggest putting an additional variable like (for example) "keyattr_dynamic_creation=True|False"

What do you think about?
Loreto

**Comment by fabiocaccamo** [comment]
@Loreton I'm glad you enjoy this library and thank you for the great feedback!

> Example.1

The behaviour your reported is a direct consequence of having the possibility to create nested dicts using the chained dotted notation `a.b.c.d.e.f` without having an `AttributeError`.

> Example.2

Attributes starting with `_` are generally **protected variables** or **dunder methods**.
Some standard libraries use/check the presence of some protected variables and methods; the side effect is that some keys are created dynamically and the resulting dictionary contains unexpected keys *(for a better understanding you could check #259)*.

I agree with the idea of adding a new option `keyattr_dynamic=True`: setting it to `False` would restrict the feature to write only already existing items otherwise an `AttributeError` is raised.

**Comment by fabiocaccamo** [comment]
@Loreton what do you think should be the default value for the `keyattr_dynamic` property?

**Comment by Loreton** [comment]
Ciao Fabio,
 I believe that the default must be exactly like for the normal dictionary: False 
So that accessing an undefined attribute gives an error. There should be no difference in results between the two access modes to the attribute value.
But this is just my thought.

**Comment by fabiocaccamo** [comment]
@Loreton the `keyattr_dynamic` property is available in `0.30.0` version.
